pkg/reconciler/v1alpha1/configuration/configuration_test.go

```diff
@@ -134,110 +135,96 @@ func TestReconcile(t *testing.T) {
+				"matching-revision-done-00001"),
 		},
 		Key: "foo/matching-revision-done",
-	}, {
-		Name: "reconcile a ready revision without the config metadata generation label",
-		Objects: []runtime.Object{
-			cfg("legacy-revision-ready", "foo", 5555,
-				WithLatestCreated,
-				WithObservedGen,
-				WithLatestReady),
-			rev("legacy-revision-ready", "foo", 5555,
-				WithCreationTimestamp(now),
-				MarkRevisionReady,
-				WithoutConfigurationMetadataGenerationLabel,
-			),
-		},
-		WantUpdates: []clientgotesting.UpdateActionImpl{{
-			Object: rev("legacy-revision-ready", "foo", 5555,
-				WithCreationTimestamp(now),
-				MarkRevisionReady,
-			),
-		}},
-		Key: "foo/legacy-revision-ready",
 	}, {
 		Name: "reconcile revision matching generation (ready: true, idempotent)",
 		Objects: []runtime.Object{
 			cfg("matching-revision-done-idempotent", "foo", 5566,
-				WithLatestCreated, WithObservedGen, WithLatestReady),
+				WithObservedGen, WithLatestCreated("matching-revision"), WithLatestReady("matching-revision")),
 			rev("matching-revision-done-idempotent", "foo", 5566,
-				WithCreationTimestamp(now), MarkRevisionReady),
+				WithCreationTimestamp(now), MarkRevisionReady, WithRevName("matching-revision")),
 		},
 		Key: "foo/matching-revision-done-idempotent",
 	}, {
 		Name: "reconcile revision matching generation (ready: false)",
 		Objects: []runtime.Object{
-			cfg("matching-revision-failed", "foo", 5555, WithLatestCreated, WithObservedGen),
+			cfg("matching-revision-failed", "foo", 5555, WithLatestCreated("matching-revision"), WithObservedGen),
 			rev("matching-revision-failed", "foo", 5555,
-				WithCreationTimestamp(now), MarkContainerMissing),
+				WithCreationTimestamp(now), MarkContainerMissing, WithRevName("matching-revision")),
 		},
 		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
 			Object: cfg("matching-revision-failed", "foo", 5555,
-				WithLatestCreated, WithObservedGen,
+				WithLatestCreated("matching-revision"), WithObservedGen,
 				// When the LatestCreatedRevision reports back a failure,
 				// then we surface that failure.
 				MarkLatestCreatedFailed("It's the end of the world as we know it")),
 		}},
 		WantEvents: []string{
 			Eventf(corev1.EventTypeWarning, "LatestCreatedFailed", "Latest created revision %q has failed",
-				"matching-revision-failed-05555"),
+				"matching-revision"),
 		},
 		Key: "foo/matching-revision-failed",
 	}, {
 		Name: "reconcile revision matching generation (ready: bad)",
 		Objects: []runtime.Object{
-			cfg("bad-condition", "foo", 5555, WithLatestCreated, WithObservedGen),
-			rev("bad-condition", "foo", 5555, WithRevStatus(v1alpha1.RevisionStatus{
-				Conditions: duckv1alpha1.Conditions{{
-					Type:     v1alpha1.RevisionConditionReady,
-					Status:   "Bad",
-					Severity: "Error",
-				}},
-			})),
+			cfg("bad-condition", "foo", 5555, WithLatestCreated("bad-condition"), WithObservedGen),
+			rev("bad-condition", "foo", 5555,
+				WithRevName("bad-condition"),
```

Comment:
This looks weird...  Not sure what to do about it :(

Reply:
yeah it's ugly - we should generally revisit this soon